### PR TITLE
refactor: app module file structure

### PR DESCRIPTION
**Pull request by itsmdasifraza**
## Description
1. Refactor the app module file structure for better organization and readability.
2. Introduce a custom theme setup for the app to enhance its visual appeal and branding.
3. Revamp the REST explorer with top priority to improve usability and user experience.
4. Implement a feature allowing HTML response preview for enhanced content visualization.

Fixes issue #737 

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [x] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [x] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

### CODE REVIEW

**Comment by punitkashyup** [comment]
Please attach the issue link

File: src/packages/@workspaces/features/rest-explorer/components/request-name/RequestName.svelte [hunk 1]
```diff
@@ -1,8 +1,9 @@
 <script lang="ts">
   import Input from "$lib/components/inputs/Input.svelte";
+  import type { UpdateRequestNameType } from "@workspaces/shared/type";
 
-  export let name;
-  export let onUpdateRequestName;
+  export let name: string;
```
Comment (gc-codes) [inline src/packages/@workspaces/features/rest-explorer/components/request-name/RequestName.svelte:5]: Comments missing everywhere for props and other places.

File: src/packages/@workspaces/shared/type/request.ts
Comment (gc-codes) [inline src/packages/@workspaces/shared/type/request.ts:1]: please add JS Docs comments.

**Review by gc-codes — CHANGES_REQUESTED** [review]
Let's add comments. It is very important. Added some more comments. Please check.

**Comment by punitkashyup** [comment]
Please attach the issue link

**Comment by punitkashyup** [comment]
Please attach the issue link

**Comment by punitkashyup** [comment]
Please attach the issue link

**Comment by punitkashyup** [comment]
Please attach the issue link

File: src/lib/components/buttons/Button.svelte [hunk 2]
```diff
@@ -78,6 +78,7 @@
     />
   {/if}
   {#if loader && !allowChild}
+    {console.log("HW")}
```
[the remark this answers is not in the document]
Comment (itsmdasifraza) [reply src/lib/components/buttons/Button.svelte]: removed

File: src/lib/components/inputs/select/Select.svelte [hunk 3]
```diff
@@ -69,12 +69,12 @@
   /**
    * Determines the background state for the Select header.
    */
-  export let headerTheme: "dark" | "transparent" = "dark";
+  export let headerTheme: "dark" | "transparent" | "voilet" = "dark";
```
[the remark this answers is not in the document]
Comment (itsmdasifraza) [reply src/lib/components/inputs/select/Select.svelte]: fixed

File: src/packages/@workspaces/features/rest-explorer/components/http-url-section/HttpUrlSection.svelte [hunk 4]
```diff
@@ -55,14 +61,19 @@
...
     borderHighlight={"active"}
     headerHighlight={"hover"}
+    minHeaderWidth={"100px"}
+    borderActiveType={"none"}
+    headerTheme={"voilet"}
     minBodyWidth={"150px"}
     zIndex={2}
+    borderType={"none"}
     menuItem={"v2"}
+    bodyTheme={"voilet"}
```
[the remark this answers is not in the document]
Comment (itsmdasifraza) [reply src/packages/@workspaces/features/rest-explorer/components/http-url-section/HttpUrlSection.svelte]: fixed

File: src/packages/@workspaces/features/rest-explorer/components/request-name/RequestName.svelte (hunk 1 above)
Comment (itsmdasifraza) [reply src/packages/@workspaces/features/rest-explorer/components/request-name/RequestName.svelte:5]: fixed

File: src/packages/@workspaces/shared/type/request.ts
Comment (itsmdasifraza) [reply src/packages/@workspaces/shared/type/request.ts:1]: added

**Review by Astitva877 — APPROVED** [review]

**Review by gc-codes — APPROVED** [review]

**Pull request merged**